actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -472,7 +472,13 @@ def initialize(parent_jar)
 
       def [](name)
         if data = @parent_jar[name.to_s]
-          parse(name, data, purpose: "cookie.#{name}") || parse(name, data)
+          result = parse(name, data, purpose: "cookie.#{name}")
+
+          unless result.nil?
+            result
+          else
+            parse(name, data)
+          end
```

Comment:
`unless` followed by `else` can be confusing, could you swap the branches:

```suggestion
          if result.nil?
            parse(name, data)
          else
            result
          end
```

Reply:
To avoid `Nested Conditionals` i prefer to check for `nil?` before overriding a value with a default:
```ruby
        if data = @parent_jar[name.to_s]
          result = parse(name, data, purpose: "cookie.#{name}")
          result = parse(name, data) if result.nil?

          result
        end
```

Reply:
I find the postfix conditional a bit harder to follow, although it's certainly subjective.